### PR TITLE
Microsoft.Solutions add 2018-02-01

### PR DESCRIPTION
**Pull request by ilahat**
Adding new version because our "patch" had required fields in the body
before. Server-side we do not allow any properties except resource-level
properties with tags being the only ones that actually get updated.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by dsgouda** [comment]
Working on network related PRs

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/managedapplications.json [hunk 1]
```diff
@@ -0,0 +1,1378 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/{applicationDefinitionId}": {
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/managedapplications.json]: This is a duplicate of line 580.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplication.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "parameters": {
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplication.json]: Parameter `applicationName` is missing.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplication.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  "responses": {
+    "200": {
+      "headers": {},
+      "body": {
+        "name": "myManagedApplication",
+        "id": "/subscriptions/subid/resourceGroups/rg/providers/Microsoft.Solutions/applications/myManagedApplication",
+        "type": "Microsoft.Solutions/applications",
+        "location": "East US 2",
+        "managedBy": null,
+        "resourceGroup": "rg",
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplication.json]: `resourceGroup` is not defined in `Application`.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplication.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        "tags": null,
+        "plan": null,
+        "identity": null,
+        "kind": "ServiceCatalog",
+        "sku":  null,
+        "properties": {
+          "managedApplicationDefinitionId": "/subscriptions/subid/resourceGroups/rg/providers/Microsoft.Solutions/managedApplicationDefinitions/myAppDef",
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplication.json]: `managedApplicationDefinitionId` is not defined in `Application`.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/createOrUpdateApplication.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      "properties": {
+        "managedApplicationDefinitionId": "/subscriptions/subid/resourceGroups/rg/providers/Microsoft.Solutions/applicationDefinitions/myAppDef",
+        "managedResourceGroupId": "/subscriptions/subid/resourceGroups/myManagedRG"
+      },
+      "name": "myManagedApplication",
+      "location": "East US 2",
+      "kind":  "ServiceCatalog"
+    }
+  },
+  "responses": {
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/createOrUpdateApplication.json]: Please also provide 200 response example.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/createOrUpdateApplication.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{
+  "parameters": {
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/createOrUpdateApplication.json]: Parameter `applicationName` missing.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/createOrUpdateApplication.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      },
+      "name": "myManagedApplication",
+      "location": "East US 2",
+      "kind":  "ServiceCatalog"
+    }
+  },
+  "responses": {
+    "201": {
+      "headers": {},
+      "body": {
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/createOrUpdateApplication.json]: Similar to getApplication.json - the response here does not match the definition of `Application`.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplicationDefinition.json [hunk 8]
```diff
@@ -0,0 +1,48 @@
+{
+  "parameters": {
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplicationDefinition.json]: Parameter `applicationDefinitionName` missing.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplicationDefinition.json [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    "subscriptionId": "subid",
+    "resourceGroupName": "rg",
+    "managedApplicationName": "myManagedApplicationDef",
+    "api-version": "2018-02-01"
+  },
+  "responses": {
+    "200": {
+      "headers": {},
+      "body": {
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplicationDefinition.json]: The response here does not match the definition of `ApplicationDefinition`.

Comment (jianghaolu) [reply specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/examples/getApplicationDefinition.json]: Same elsewhere.

File: specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/managedapplications.json [hunk 10]
```diff
@@ -0,0 +1,1378 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns the managed application.",
+            "schema": {
+              "$ref": "#/definitions/Application"
+            }
+          },
+          "404": {
```
Comment (jianghaolu) [inline specification/resources/resource-manager/Microsoft.Solutions/stable/2018-02-01/managedapplications.json]: All the negative response status code descriptions MUST be defined in the description of the "default" status code (if the intention is for documentation purpose only). Example: "description": "'404' - 'Not found'.\n'400' - 'Bad request'.\n'500': 'Internal Server Error'."' 

Same elsewhere.

**Review by jianghaolu — CHANGES_REQUESTED** [review]

**Comment by veronicagg** [comment]
@ilahat should this PR be closed? it's been opened for a long time with no activity. thanks.

**Comment by jianghaolu** [comment]
Closing as we haven't heard back. Please feel free to reopen, thanks!

**Pull request closed without merging**